File: app/src/main/java/com/playgilround/schedule/client/calendarschedule/MainActivity.java
package com.playgilround.schedule.client.calendarschedule;

import android.annotation.SuppressLint;
import android.os.Bundle;
import androidx.viewpager.widget.ViewPager;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import android.view.animation.Animation;
import android.view.animation.Transformation;
import android.widget.LinearLayout;
import android.widget.ScrollView;

import com.playgilround.schedule.client.calendarschedule.adapter.ScheduleAdapter;
import com.playgilround.schedule.client.calendarschedule.view.CalendarView;
import com.playgilround.schedule.client.calendarschedule.view.ExpandIconView;
import com.playgilround.schedule.client.calendarschedule.view.ScheduleRecyclerView;

public class MainActivity extends AppCompatActivity {


    private boolean expanded = true;

    static final String TAG = MainActivity.class.getSimpleName();

    private int mInitHeight = 0;

    //State
    public static final int STATE_EXPANDED = 0;
    public static final int STATE_COLLAPSED = 1;
    CalendarView calendarView;
    ViewPager mPageView;

    ScrollView scrollView;

    protected ExpandIconView expandIconView;
    private ScheduleRecyclerView rvScheduleList;


    private int mState = STATE_EXPANDED;

    private ScheduleAdapter mScheduleAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        calendarView = findViewById(R.id.calendarView);
        scrollView = findViewById(R.id.scrollView);
        mPageView = findViewById(R.id.calendarViewPager);
        expandIconView = findViewById(R.id.expandIcon);
        rvScheduleList = findViewById(R.id.rvScheduleList);

        ExpandIconView expandIconView = findViewById(R.id.expandIcon);

        expandIconView.setState(ExpandIconView.LESS, true);

        expandIconView.setOnClickListener(l -> {
            if (expanded) {
                reduction(1000);
                expanded = false;

            } else {
                expansion(1000);
                expanded = true;
            }
        });

        initScheduleList();
    }

    @SuppressLint("WrongConstant")
    private void initScheduleList() {
        LinearLayoutManager manager = new LinearLayoutManager(this);
        manager.setOrientation(LinearLayoutManager.VERTICAL);

        rvScheduleList.setLayoutManager(manager);
        DefaultItemAnimator itemAnimator = new DefaultItemAnimator();
        itemAnimator.setSupportsChangeAnimations(false);
        rvScheduleList.setItemAnimator(itemAnimator);

        mScheduleAdapter = new ScheduleAdapter(this);
        rvScheduleList.setAdapter(mScheduleAdapter);
    }

    @Override
    public void onWindowFocusChanged(boolean hasFocus) {

        mInitHeight = calendarView.getHeight();
    }

    //캘린더 축소
    public void reduction(int duration) {
        if (getState() == STATE_EXPANDED) {
            final int currentHeight = mInitHeight;
            final int targetHeight = 166;
            final int topHeight = 166;

            Animation anim = new Animation() {
                @Override
                protected void applyTransformation(float interpolatedTime, Transformation t) {
                    scrollView.getLayoutParams().height = (interpolatedTime == 1)
                            ? targetHeight
                            : currentHeight - (int) ((currentHeight - targetHeight) * interpolatedTime);
                    scrollView.requestLayout();

                    if (scrollView.getMeasuredHeight() < topHeight + targetHeight) {
                        int position = topHeight + targetHeight - scrollView.getMeasuredHeight();
                        scrollView.smoothScrollTo(0, position);
                    }

                    if (interpolatedTime == 1) {
                        setState(STATE_COLLAPSED);
                    }
                }
            };
            anim.setDuration(duration);
            scrollView.startAnimation(anim);
        }
        expandIconView.setState(ExpandIconView.MORE, true);
    }

    //캘린더 확장
    public void expansion(int duration) {
        if (getState() == STATE_COLLAPSED) {

            final int currentHeight = 166;
            final int targetHeight = mInitHeight;

            Animation anim = new Animation() {
                @Override
                protected void applyTransformation(float interpolatedTime, Transformation t) {
                    scrollView.getLayoutParams().height = (interpolatedTime == 1)
                            ? LinearLayout.LayoutParams.WRAP_CONTENT
                            : currentHeight - (int) ((currentHeight - targetHeight) * interpolatedTime);
                    scrollView.requestLayout();

                    if (interpolatedTime == 1) {
                        setState(STATE_EXPANDED);
                    }
                }
            };
            anim.setDuration(duration);
            scrollView.startAnimation(anim);

        }
        expandIconView.setState(ExpandIconView.LESS, true);
    }


    public int getState() {
        return mState;
    }


    public void setState(int state) {
        this.mState = state;
    }

}

File: app/src/main/java/com/playgilround/schedule/client/calendarschedule/MainActivity.kt
package com.playgilround.schedule.client.calendarschedule

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.view.animation.Transformation
import android.widget.LinearLayout
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.android.material.bottomsheet.BottomSheetBehavior.STATE_COLLAPSED
import com.google.android.material.bottomsheet.BottomSheetBehavior.STATE_EXPANDED
import com.playgilround.schedule.client.calendarschedule.R.id.*
import com.playgilround.schedule.client.calendarschedule.adapter.ScheduleAdapter
import com.playgilround.schedule.client.calendarschedule.view.ExpandIconView
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.calendar_view.*
import kotlinx.android.synthetic.main.schedule_view.*
import java.util.*

class MainActivity : AppCompatActivity() {

    private var expanded : Boolean = true

    var mInitHeight : Int = 0
    var mState = STATE_EXPANDED

    private lateinit var mScheduleAdapter: ScheduleAdapter

    private lateinit var fabOpen: Animation
    private lateinit var fabClose: Animation

    private var isFabOpen = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        fabOpen = AnimationUtils.loadAnimation(applicationContext, R.anim.floating_open)
        fabClose = AnimationUtils.loadAnimation(applicationContext, R.anim.floating_close)

        floatingBtn.setOnClickListener {
            isFabOpen = if (isFabOpen) {
                floatingBtn.setImageResource(R.drawable.exit)
                floatingAdd.startAnimation(fabClose)
                floatingModify.startAnimation(fabClose)
                floatingDelete.startAnimation(fabClose)
                false
            } else {
                floatingBtn.setImageResource(R.drawable.add)
                floatingAdd.startAnimation(fabOpen)
                floatingModify.startAnimation(fabOpen)
                floatingDelete.startAnimation(fabOpen)
                true
            }
        }

       /* floatingBtn.setOnClickListener {
            floatingBtn.setImageResource(R.drawable.exit)
            if (calendarView.selectedDates.size == 0) {
                Toast.makeText(applicationContext, getString(R.string.toast_please_select_day), Toast.LENGTH_SHORT).show()
            } else {
                val iterator = (calendarView.selectedDates).iterator()
                iterator.forEach {
                    Log.d(TAG, "calendar time ->" + it.time.toString())
                }
            }
        }*/
        initScheduleList()
    }

    @SuppressLint("WrongConstant")
    private fun initScheduleList() {
        val manager = LinearLayoutManager(this)
        manager.orientation = LinearLayoutManager.VERTICAL

        rvScheduleList!!.layoutManager = manager

        val itemAnimation = DefaultItemAnimator()
        itemAnimation.supportsChangeAnimations = false
        rvScheduleList!!.itemAnimator = itemAnimation

        mScheduleAdapter = ScheduleAdapter(this)
        rvScheduleList!!.adapter = mScheduleAdapter
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        mInitHeight = calendarView.height
    }

    companion object {
        val STATE_EXPANDED = 0
        val STATE_COLLAPSED = 1

        val TAG = MainActivity.javaClass.simpleName;
    }

}